### PR TITLE
Move tx_hashes_to_block out of TxManager

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

`tx_hashes_to_block` currently lives inside `TxManager`, which is the object that deals with tracking `Tx`s (the original transaction type, as opposed to the newer mint-related transactions). During the work on adding minting this function was repurposed to get a list of values, that are now either a `Tx`, or a `MintTx` or a `MintConfigTx` and then use the consensus enclave to form a block from them.
This is hacky, and gets in the way of follow-up work on `MintTx` verification that we want to do - specifically, we want to have the enclave validate `MintTx`s against active `MintConfigTx`s, which means the active `MintConfigTx` information needs to make its way to the enclave. `TxManager` has no access to that, and while it could be bolted on that is wrong and repurposing the original purpose of that object.

I think the right thing to do is to change `tx_hashes_to_block` to only deal with `Tx`s, as was originally intended, and move the actual block forming that combines all transaction types into `ByzantineLedgerWorker`, which is where we go from SCP externalized values into a block formed by the enclave and then into the ledger database.

### In this PR
* Add a method `form_block_from_externalized_values` to `ByzantineLedgerWorker` that deals with forming a new block using the enclave and inputs externalized from SCP
* Rename `tx_hashes_to_block` to `tx_hashes_to_well_formed_encrypted_txs_and_proofs` and remove block-forming functionality from it
* Test fixes/shuffling
* Remove no longer used method `get_root_tx_out_membership_element` from `TxManagerUntrustedInterfaces`

### CODE REVIEW

File: consensus/service/src/byzantine_ledger/worker.rs [hunk 1]
```diff
@@ -795,6 +805,70 @@ impl<
...
+        externalized_values: Vec<ConsensusValue>,
+    ) -> BlockData {
+        let num_blocks = self
+            .ledger
+            .num_blocks()
+            .expect("Ledger must contain a block.");
+        let parent_block = self
+            .ledger
+            .get_block(num_blocks - 1)
+            .expect("Ledger must contain a block.");
```
Comment (remoun) [inline consensus/service/src/byzantine_ledger/worker.rs:820]: Optional/Equivalent: Change this message to make it easier to differentiate which call failed.

```suggestion
            .expect("Failed to get last block.");
```

**Review by remoun — APPROVED** [review]
LGTM

File: consensus/service/src/byzantine_ledger/worker.rs [hunk 2]
```diff
@@ -795,6 +805,70 @@ impl<
...
+        externalized_values: Vec<ConsensusValue>,
+    ) -> BlockData {
+        let num_blocks = self
+            .ledger
+            .num_blocks()
+            .expect("Ledger must contain a block.");
+        let parent_block = self
+            .ledger
+            .get_block(num_blocks - 1)
+            .expect("Failed to get last block.");
```
Comment (remoun) [inline consensus/service/src/byzantine_ledger/worker.rs:820]: Optional/Equivalent: Use `get_latest_block`
```suggestion
        let parent_block = self
            .ledger
            .get_latest_block()
            .expect("Failed to get latest block.");
```

**Review by remoun — APPROVED** [review]

**Review by iamalwaysuncomfortable — APPROVED** [review]
This was a good idea, this is a much more logical place to put it.

**Pull request merged**